src/Methods/BuilderHelper.php

```diff
@@ -30,6 +30,9 @@ class BuilderHelper
     /** @var string[] */
     public const MODEL_CREATION_METHODS = ['make', 'create', 'forceCreate', 'findOrNew', 'firstOrNew', 'updateOrCreate', 'fromQuery', 'firstOrCreate'];
 
+    /** @var string[] */
+    public const COLLECTION_RETRIEVAL_METHODS = ['hydrate', 'findMany', 'get'];
```

Comment:
I think this is not needed. We can solve it in a more general way. In other comments, I will try to describe how might we do it.